### PR TITLE
Add Cypress tests and documentation

### PR DESCRIPTION
**Pull request by chutch1122**
This PR addresses some things requested by the maintainer of NGINX Proxy Manager

- Cypress automated tests
- Documentation for setting up SSO with OIDC
- A more user-friendly error message for when a user does not exist in NPM and they are attempting to sign in via an IdP
- Fixing a typo in the "hint" description so that the name of "Redirect URL" matches the field label in the modal.

### CODE REVIEW

File: docs/src/setup/index.md [hunk 1]
```diff
@@ -146,4 +146,38 @@ Immediately after logging in with this default user you will be asked to modify
       INITIAL_ADMIN_PASSWORD: mypassword1
 ```
 
+## OpenID Connect - Single Sign-On (SSO)
+
+Nginx Proxy Manager supports single sign-on (SSO) with OpenID Connect. This feature allows you to use an external OpenID Connect provider log in.
+*Note: This feature requires a user to have an existing account to have been created via the "Users" page in the admin interface.*
```
Comment (oechsler) [inline docs/src/setup/index.md]:
```suggestion

::: warning

Please note, that this feature requires a user to have an existing account to have been created via the "Users" page in the admin interface.

:::
```

A small suggestion: include your note in the warning block for consistency with the rest of the document and better visibility on the rendered page.

**Review by oechsler — APPROVED** [review]

File: docs/src/setup/index.md (hunk 1 above)
Comment (chutch1122) [reply docs/src/setup/index.md]: Done!

**Comment by chutch1122** [comment]
@oechsler I just added some additional UI-based end-to-end tests. These tests test logging in with a test user when OIDC is both enabled and disabled

**Comment by oechsler** [comment]
Hey @chutch1122, I completely missed the fact that you made further changes with your last push and merged them directly. If you think a review would still be beneficial, I can revert the changes and reopen this PR. Apologies for the mix-up on my part.

**Comment by chutch1122** [comment]
@oechsler Feel free to take a look at it. I'm just opened up a new PR anyway to merge in changes from the upstream `develop` branch. There were some changes to error messages at some point after the last time `develop` was pulled in that was causing the build to fail

**Pull request merged**